client/src/test/scala/examples/ScalaSparkClientUsage.scala

```diff
@@ -142,7 +142,7 @@ object ScalaSparkClientUsage extends App {
     y <- yPromise.future
   } yield (complexMath(x, y))
 
-  resultPromise.onSuccess {case x => println(s"Added result is ${x}") }
+  resultPromise.onComplete { x => println(s"Added result is ${x}") }
```

Comment:
why this change? is this related to 2.12?

Reply:
Indeed, onSuccess is deprecated in Scala 2.12 whereas onComplete is fine for both 2.11/12